src/lib/bootloader/bls.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require "fileutils"
+require "yast"
+require "bootloader/sysconfig"
+require "bootloader/cpu_mitigations"
+require "cfa/grub2/default"
+
+Yast.import "Report"
+
+module Bootloader
+  # Represents bls compatile system calls which can be used
+  # e.g. by grub2-bls and systemd-boot
+  class Bls
+    include Yast::Logger
+    extend Yast::I18n
+
+    SDBOOTUTIL = "/usr/bin/sdbootutil"
+
+    def initialize
+      textdomain "bootloader"
+    end
+
+    def self.create_menu_entries
+      Yast::Execute.on_target!(SDBOOTUTIL, "--verbose", "add-all-kernels")
+    rescue Cheetah::ExecutionFailed => e
+      Yast::Report.Error(
+        format(_(
+                 "Cannot create boot menu entry:\n" \
+                 "Command `%{command}`.\n" \
+                 "Error output: %{stderr}"
+               ), command: e.commands.inspect, stderr: e.stderr)
+      )
+    end
+
+    def self.install_bootloader
+      Yast::Execute.on_target!(SDBOOTUTIL, "--verbose",
+        "install")
+    rescue Cheetah::ExecutionFailed => e
+      Yast::Report.Error(
+      format(_(
+               "Cannot install bootloader:\n" \
+               "Command `%{command}`.\n" \
+               "Error output: %{stderr}"
+             ), command: e.commands.inspect, stderr: e.stderr)
+    )
+    end
+
+    def self.write_menu_timeout(timeout)
+      Yast::Execute.on_target(SDBOOTUTIL, "set-timeout", timeout)
```

Comment:
here and also below this code it is using non`!` varient. It means it will return nil and not raise exception, but the code below expect it. See https://github.com/yast/yast-yast2/blob/717d99f214a56645216bb9c5f4f32d1244d4fe35/library/system/src/lib/yast2/execute.rb#L69 for diff between `on_target` and `on_target!`

Reply:
Ups, yes you are right ! Fixed.